### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by wcbeard**
added airflow functions and test

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,4 +15,5 @@
     license=["Apache 2.0", "MIT"],
     extras_require={"test": ["pytest"]},
     classifiers=["Programming Language :: Python :: 3.6"],
+    download_url="https://github.com/wcbeard/bgbb_lib/archive/v0.1.2.tar.gz",
```
Comment (acmiyaguchi) [inline setup.py]: This is going to change on every release, it might be easier just to release via the Github UI.

File: tests/test_job_fns.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    def gen_client_days(
+        client: dict, day_range, p: float, θ: float, ensure_first=True
+    ):
+        """If `ensure_first`, add 1st day of day_range to their history
+        so that every client will show up in `rfn`.
+        """
+        days_used_browser = []
+        for day in day_range:
+            # die coin
+            if coin_flip(θ):
```
Comment (acmiyaguchi) [inline tests/test_job_fns.py]: θ is a tough one to type.

File: tests/test_job_fns.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        nr.seed(seed)
+        table_data = gen_client_dicts(
+            n_clients_in_sample=n_clients_in_sample, abgd=abgd
+        )
+
+        dataframe = cdaily_factory(table_data)
+        dataframe.createOrReplaceTempView("clients_daily")
+        dataframe.cache()
+        return dataframe
+        # yield dataframe
```
Comment (acmiyaguchi) [inline tests/test_job_fns.py]: Remove extra comments.

File: tests/test_job_fns.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            if not cid:
+                print(cid_rows)
+            cids_rows.extend(cid_rows)
+        return cids_rows
+
+    cdaily_factory = generate_data(dataframe_factory)
+
+    # @fixture
+    def gen_clients_daily(n_clients_in_sample, abgd=[1, 3, 1, 10], seed=0):
+        nr.seed(seed)
```
Comment (acmiyaguchi) [inline tests/test_job_fns.py:105]: Does this seed function get called before the fixtures are generated? It's probably a good idea for these probabilistic tests to be deterministic, although the test case looks robust.

**Review by acmiyaguchi — APPROVED** [review]
It looks like the main entry-point is missing, but this looks like a solid foundation.

File: tests/test_job_fns.py (hunk 4 above)
Comment (wcbeard) [reply tests/test_job_fns.py:105]: The seed should be set at runtime (when gen_clients_daily is called), but don't know enough about pytest to know if that's while or after the fixture is generated.

I was concerned with generating realistic data to not have to worry about breaking the fitting process (which would also allow more samples).

**Review by acmiyaguchi — APPROVED** [review]
Looks good to me.

**Pull request merged**